src/main/java/io/quarkus/search/app/hibernate/AnalysisConfigurer.java

```diff
@@ -49,88 +50,90 @@ public void configure(ElasticsearchAnalysisConfigurationContext context) {
         EnumSet<Language> englishLanguages = EnumSet.of(Language.ENGLISH, Language.PORTUGUESE, Language.SPANISH);
 
         for (Language language : englishLanguages) {
-            SharedFilters result = sharedFilters(context, language);
-
-            context.analyzer(defaultAnalyzer(language)).custom()
-                    .tokenizer("standard")
-                    .tokenFilters(result.compoundTechnicalNameFilter(), "lowercase", result.possessiveStemmer(), result.stop(),
-                            result.regularStemmer(), "asciifolding")
-                    .charFilters("html_strip");
-            context.analyzer(defaultSearchAnalyzer(language)).custom()
-                    .tokenizer("standard")
-                    // > In general, synonym filters rewrite their inputs to the tokenizer and filters used in the preceding analysis chain
-                    // Note how the synonym filter is added in the end. According to https://www.elastic.co/blog/boosting-the-power-of-elasticsearch-with-synonyms
-                    // preceding filters should get applied to the synonyms we passed to it, so we don't need to bother about normalizing them in some way:
-                    .tokenFilters("lowercase", result.possessiveStemmer(), result.stop(),
-                            result.regularStemmer(),
+            SharedFilters sharedFilters = sharedFilters(context, language);
+
+            new LanguageAnalyzerConfigurator(language, sharedFilters)
```

Comment:
I think it will be very confusing if you introduce new types that are not included in Hibernate Search... Ideally `SharedFilters` and `LanguageAnalyzerConfigurator` would have to go.

Isn't there a way to make this work simply with methods? That way you'd clearly separate each part, while still showing the real APIs from Hibernate Search.

I.e.:

```
void configureEnglishLikeLanguage(ElasticsearchAnalysisConfigurationContext context, Language language) {
    configureSharedFilters(context, language);
    context.analyzer(defaultAnalyzer(language)).custom()
    ...
}


void configureJapanese(ElasticsearchAnalysisConfigurationContext context, Language language) {
    configureSharedFilters(context, language);
   
    context.analyzer(defaultAnalyzer(language)).custom()
    ...
}
```